x-pack/plugins/observability_solution/dataset_quality/public/components/dataset_quality_details/degraded_field_flyout/possible_mitigations/field_limit/increase_field_mapping_limit.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFieldText,
+  EuiFormRow,
+  EuiButton,
+  EuiFieldNumber,
+} from '@elastic/eui';
+import {
+  fieldLimitMitigationApplyButtonText,
+  fieldLimitMitigationCurrentLimitLabelText,
+  fieldLimitMitigationNewLimitButtonText,
+  fieldLimitMitigationNewLimitPlaceholderText,
+} from '../../../../../../common/translations';
+import { useDegradedFields } from '../../../../../hooks';
+
+export function IncreaseFieldMappingLimit({ totalFieldLimit }: { totalFieldLimit: number }) {
+  // Propose the user a 30% increase over the current limit
+  const proposedNewLimit = Math.round(totalFieldLimit * 1.3);
+  const [newFieldLimit, setNewFieldLimit] = useState<number>(proposedNewLimit);
+  const [isInvalid, setIsInvalid] = useState(false);
+  const { updateNewFieldLimit, isSavingNewFieldLimitInProgress } = useDegradedFields();
+
+  const validateNewLimit = (newLimit: string) => {
+    const parsedLimit = parseInt(newLimit, 10);
+    setNewFieldLimit(parsedLimit);
+    if (totalFieldLimit > parsedLimit) {
+      setIsInvalid(true);
+    } else {
+      setIsInvalid(false);
+    }
+  };
+
+  return (
+    <EuiFlexGroup gutterSize="s">
+      <EuiFlexItem>
+        <EuiFormRow label={fieldLimitMitigationCurrentLimitLabelText}>
+          <EuiFieldText
+            data-test-subj="datasetQualityIncreaseFieldMappingLimitFieldText"
+            disabled
+            value={totalFieldLimit}
+          />
+        </EuiFormRow>
+      </EuiFlexItem>
+      <EuiFlexItem>
+        <EuiFormRow label={fieldLimitMitigationNewLimitButtonText}>
+          <EuiFieldNumber
+            data-test-subj="datasetQualityIncreaseFieldMappingLimitFieldText"
+            placeholder={fieldLimitMitigationNewLimitPlaceholderText}
+            value={newFieldLimit}
+            onChange={(e) => validateNewLimit(e.target.value)}
+            aria-label={fieldLimitMitigationNewLimitPlaceholderText}
+            isInvalid={isInvalid}
+            min={totalFieldLimit + 1}
+          />
+        </EuiFormRow>
+      </EuiFlexItem>
+      <EuiFlexItem grow={false}>
+        <EuiFormRow hasEmptyLabelSpace>
```

Comment:
> [!TIP]
> You can remove this extra node as per doc is not necessary if there are no additional label/helpText on the control.

Reply:
This node with this specific class `hasEmptyLabelSpace` makes sure that the label space is preserved when it is not present. If you remove this, it will cause misalignment

<img width="430" alt="image" src="https://github.com/user-attachments/assets/e28d0550-b3b3-4391-a698-ac34c3198652">